ocaml/xapi/repository.ml

```diff
@@ -543,7 +543,24 @@ let get_repository_handler (req : Http.Request.t) s _ =
         match String.sub_to_end req.Request.uri len with
         | uri_path ->
             let root = !Xapi_globs.local_pool_repo_dir in
-            Fileserver.response_file s (Helpers.resolve_uri_path ~root ~uri_path)
+            let uri_path' =
+              match
+                String.starts_with ~prefix:Constants.enabled_repo_uri_prefix
+                  uri_path
+              with
+              | true ->
+                  let repo = get_single_enabled_update_repository ~__context in
+                  let repo_name =
+                    get_remote_repository_name ~__context ~self:repo
```

Comment:
A case like `CDN -> pool A -> pool B -> pool C` needs to be considered.
E.g. pool B will get updates from pool A. In this case, the origin or the repo in pool A is neither `remote` nor `bundle`.

Reply:
```
let get_remote_repository_name ~__context ~self =
  let prefix =
    match Db.Repository.get_origin ~__context ~self with
    | `remote ->
        !Xapi_globs.remote_repository_prefix
    | `bundle ->
        !Xapi_globs.bundle_repository_prefix
    | `remote_pool ->
        !Xapi_globs.remote_pool_repository_prefix
  in
  prefix ^ "-" ^ get_repository_name ~__context ~self
```
In the feature branch, `get_remote_repository_name` has already included `remote_pool` case.